packages/database/exp/register.ts

```diff
@@ -38,7 +38,17 @@ export function registerDatabase(variant?: string): void {
       (container, { instanceIdentifier: url }) => {
         const app = container.getProvider('app-exp').getImmediate()!;
         const authProvider = container.getProvider('auth-internal');
-        return repoManagerDatabaseFromApp(app, authProvider, url);
+        // App check is unsupported in Node.
+        const appCheckProvider =
+          variant === 'node'
+            ? undefined
+            : container.getProvider('app-check-internal');
```

Comment:
I think in practice it doesn't make any difference because `appCheckProvider.get()` will not resolve in Nodejs and database will just work without AppCheck. And since it is a shared code, I would actually prefer to not have the special handling for nodejs.

Reply:
I would also prefer that we simplify this again and either call `container.getProvider('app-check')` directly and rely on the fact that Node will return `undefined` here or pass a "isNode" boolean into registerDatabase (with different values from index.ts and index.node.ts). 

The question is though if the debug AppCheck will ever work in Node. If that is the case then we should just let `container` decide if AppCheck is available.

Reply:
Reverted to simple container.getProvider.

Reply:
Debug AppCheck can work in Node, we just need to read from `globalThis` ( need a polyfill) instead of `self`. I think it'll be useful for server side rendering, though it sounds like an abuse of the debug token.
Also custom providers should already be working in Nodejs.

Therefore, I think we should 
1. Read debug token from `globalThis`. We need to define a polyfill for `globalThis`, probably in `@firebase/util`. Here is a [reference implementation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis#search_for_the_global_across_environments). 
2. Always pass the AppCheckProvider database

Reply:
Ok, will make a separate PR for Node support after merge. Reverting other Node-related change above.